crates/sc-proof-of-time/src/gossip.rs

```diff
@@ -126,11 +126,17 @@ impl<Block: BlockT> Validator<Block> for PotGossipValidator {
     fn validate(
         &self,
         _context: &mut dyn ValidatorContext<Block>,
-        _sender: &PeerId,
+        sender: &PeerId,
         mut data: &[u8],
     ) -> ValidationResult<Block::Hash> {
         match PotProof::decode(&mut data) {
-            Ok(_) => ValidationResult::ProcessAndKeep(topic::<Block>()),
+            Ok(proof) => {
+                if self.pot_state.is_candidate(*sender, &proof).is_ok() {
```

Comment:
1) I'd have some trace level logging here indicating the reason of rejection
2) It will still forward invalid messages, meaning receiver that validates and sees that message is no good will ban peer that forwarded the message, not originator; my expectation was that we do full validation here and do not let through messages that are invalid at all